core/domain/topic_domain.py

```diff
@@ -580,6 +580,30 @@ def require_valid_name(cls, name):
                 'Topic name should be at most %d characters, received %s.'
                 % (name_limit, name))
 
+    @classmethod
+    def require_valid_abbreviated_name(cls, name):
+        """Checks whether the abbreviated name of the topic is a valid one.
+
+        Args:
+            name: str. The abbreviated name to validate.
+        """
+        if not isinstance(name, python_utils.BASESTRING):
+            raise utils.ValidationError('Abbreviated name should be a string.')
+
+        if name == '':
+            raise utils.ValidationError(
+                'Abbreviated name field should not be empty.')
+        if '-' in name:
+            raise utils.ValidationError(
+                'Abbreviated name field should not contain hyphens.')
```

Comment:
Perhaps "abbreviated name" is the wrong word, if we mean something like "topic url fragment".